### PR TITLE
DEV010 Create controleer for musician skills model

### PR DESCRIPTION
**Pull request by YanPam**
create controleer for musican scills model
[card](https://trello.com/c/rFzbSbxb/38-dev010-create-controller-for-musician-skills-model)

### CODE REVIEW

File: app/assets/javascripts/musician_skills.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (andreydro) [inline app/assets/javascripts/musician_skills.coffee]: Delete empty file

File: app/assets/stylesheets/musician_skills.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the MusicanScillsController controller here.
```
Comment (andreydro) [inline app/assets/stylesheets/musician_skills.scss]: Delete empty file

File: app/helpers/musician_skills_helper.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module MusicianSkillsHelper
```
Comment (andreydro) [inline app/helpers/musician_skills_helper.rb]: Delete empty file

File: db/schema.rb [hunk 4]
```diff
@@ -10,7 +10,7 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2019_08_08_171446) do
```
Comment (andreydro) [inline db/schema.rb]: In that tasks you don't have migrations, so your schema.rb should not change

File: spec/controllers/musician_skills_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
```
Comment (andreydro) [inline spec/controllers/musician_skills_controller_spec.rb]: get rid of that require

File: spec/helpers/musician_skills_helper_spec.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
```
Comment (andreydro) [inline spec/helpers/musician_skills_helper_spec.rb]: delete empty file

File: app/controllers/api/musician_skills_controller.rb [hunk 7]
```diff
@@ -0,0 +1,46 @@
+module Api
+  class MusicianSkillsController < ApplicationController
+    before_action :set_musican_skill, only: %i[show update destroy]
```
Comment (als6k) [inline app/controllers/api/musician_skills_controller.rb]: set_musician_skill

File: app/controllers/api/musician_skills_controller.rb [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+    def index
+      @musican_skill = MusicianSkill.all.order('created_at DESC')
```
Comment (als6k) [inline app/controllers/api/musician_skills_controller.rb]: @musician_skill
Change name of variable for correct.

File: app/controllers/api/musician_skills_controller.rb [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+      render json: @musican_skill
```
Comment (als6k) [inline app/controllers/api/musician_skills_controller.rb]: @musician_skill

File: app/controllers/api/musician_skills_controller.rb [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+    end
+
+    def show
+      render json: @musican_skill
```
Comment (als6k) [inline app/controllers/api/musician_skills_controller.rb]: @musician_skill

File: app/controllers/api/musician_skills_controller.rb [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+    end
+
+    def create
+      @musican_skill = MusicianSkill.new(skill_params)
```
Comment (als6k) [inline app/controllers/api/musician_skills_controller.rb]: @musician_skill

File: app/controllers/api/musician_skills_controller.rb [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+      if @musican_skill.save
```
Comment (als6k) [inline app/controllers/api/musician_skills_controller.rb]: @musician_skill

File: app/controllers/api/musician_skills_controller.rb [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+        render json: @musican_skill, status: :created
```
Comment (als6k) [inline app/controllers/api/musician_skills_controller.rb]: @musician_skill

File: app/controllers/api/musician_skills_controller.rb [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+      else
+        render json: @musican_skill.errors, status: :unprocessable_entity
```
Comment (als6k) [inline app/controllers/api/musician_skills_controller.rb]: @musician_skill

File: app/controllers/api/musician_skills_controller.rb [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+      end
+    end
+
+    def update
+      if @musican_skill.update(skill_params)
```
Comment (als6k) [inline app/controllers/api/musician_skills_controller.rb]: @musician_skill

File: app/controllers/api/musician_skills_controller.rb [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+        render json: @musican_skill, status: :ok
```
Comment (als6k) [inline app/controllers/api/musician_skills_controller.rb]: @musician_skill

File: app/controllers/api/musician_skills_controller.rb [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+      else
+        render json: @musican_skill.errors, status: :unprocessable_entity
```
Comment (als6k) [inline app/controllers/api/musician_skills_controller.rb]: @musician_skill

File: app/controllers/api/musician_skills_controller.rb [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+      end
+    end
+
+    def destroy
+      @musican_skill.delete
```
Comment (als6k) [inline app/controllers/api/musician_skills_controller.rb]: @musician_skill

File: app/controllers/api/musician_skills_controller.rb [hunk 19]
```diff
@@ -0,0 +1,46 @@
...
+      head :no_content
+    end
+
+    private
+
+    def set_musican_skill
```
Comment (als6k) [inline app/controllers/api/musician_skills_controller.rb]: set_musician_skill

File: app/controllers/api/musician_skills_controller.rb [hunk 20]
```diff
@@ -0,0 +1,46 @@
...
+      @musican_skill = User.find(params[:id])
```
Comment (als6k) [inline app/controllers/api/musician_skills_controller.rb]: @musician_skill
WTF??

File: app/controllers/api/musician_skills_controller.rb [hunk 21]
```diff
@@ -0,0 +1,46 @@
...
+      render json: @musician_skills
+    end
+
+    def show
+      render json: @musician_skill
+    end
+
+    def create
+      @musician_skill = MusicianSkill.new(skill_params)
+      if @musician_skill.save
```
Comment (andreydro) [inline app/controllers/api/musician_skills_controller.rb]: `save!`

File: app/controllers/api/musician_skills_controller.rb [hunk 22]
```diff
@@ -0,0 +1,46 @@
...
+        render json: @musician_skill, status: :created
+      else
+        render json: @musician_skill.errors, status: :unprocessable_entity
+      end
+    end
+
+    def update
+      if @musician_skill.update(skill_params)
```
Comment (andreydro) [inline app/controllers/api/musician_skills_controller.rb]: `update!`

File: app/controllers/api/musician_skills_controller.rb [hunk 23]
```diff
@@ -0,0 +1,46 @@
...
+        render json: @musician_skill, status: :ok
+      else
+        render json: @musician_skill.errors, status: :unprocessable_entity
+      end
+    end
+
+    def destroy
+      @musician_skill.delete
```
Comment (andreydro) [inline app/controllers/api/musician_skills_controller.rb]: what id delete fails?

File: db/schema.rb [hunk 24]
```diff
@@ -34,8 +33,8 @@
   end
 
   create_table "images", force: :cascade do |t|
-    t.integer "imageable_id", null: false
```
Comment (andreydro) [inline db/schema.rb]: you should not have this changes in here

File: app/controllers/api/musician_skills_controller.rb [hunk 25]
```diff
@@ -0,0 +1,46 @@
...
+      head :no_content
+    end
+
+    private
+
+    def set_musician_skill
+      @musician_skill = MusicianSkill.find(params[:id])
```
Comment (andreydro) [inline app/controllers/api/musician_skills_controller.rb]: get id from `skill_params`

File: db/seeds.rb [hunk 26]
```diff
@@ -1 +1 @@
-User.create(nickname: 'eqe', first_name: 'qwe', second_name: 'quno', type: 'User', email: 'rynf', password: 'tntfm')
\ No newline at end of file
+User.create(nickname: 'eqe', first_name: 'qwe', second_name: 'quno', type: 'User', email: 'rynf', password: 'tntfm')
```
Comment (andreydro) [inline db/seeds.rb]: Use gem faker to generate content for seeds

File: app/serializers/musician_skill_serializer.rb [hunk 27]
```diff
@@ -0,0 +1,17 @@
+class MusicianSkillSerializer < ActiveModel::Serializer
```
[the remark this answers is not in the document]
Comment (andreydro) [reply app/serializers/musician_skill_serializer.rb]: +1

File: app/controllers/api/musician_skills_controller.rb [hunk 28]
```diff
@@ -0,0 +1,49 @@
...
+    end
+
+    private
+
+    def set_musician_skill
+      @musician_skill = MusicianSkill.find(skill_params[:id])
+    end
+
+    def skill_params
+      params.require(:musician_skill).permit(:name)
```
Comment (andreydro) [inline app/controllers/api/musician_skills_controller.rb]: If you do `skill_params[:id]` then your skill_params should have `id`

**Pull request merged**